state/api/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityView.java

```diff
@@ -25,6 +26,25 @@
 
 /**
  * Provides a unified view of the current state of all entities held in Stream Registry.
+ * <p/>
+ * The underlying state storage from which this is this is loaded may support entities which have been deleted. If
+ * that is the case it is important to understand the lifecycle of entities as it relates to {@link EntityView}.
+ * <p/>
+ * Entities may be created and updated in the underlying state storage. Any entity which has been created (but not
+ * deleted) will be returned by {@link #all(Class)} and {@link #get(Key)}. It is expected that users handle said
+ * entities accordingly (update external dependencies/respond to view requests etc).
+ * <p/>
+ * When entities are deleted in the underlying state storage, they will stop being returned by {@link #all(Class)} and
+ * {@link #get(Key)}. Instead they will be returned by {@link #allDeleted(Class)}. In this case it is expected that
+ * users handle those delete requests accordingly (e.g. remove the entity from external dependencies). This handling
+ * does not automatically remove the entities from this in memory state (or the underlying state storage). In order to
+ * remove the entity, {@link #purgeDeleted(Key)} should be used. This will remove all in memory references to said
+ * entity and it will no longer be returned by {@link #allDeleted(Class)}.
```

Comment:
Line length/wrapping needs fixing, but here's a suggestion for the wording:
```suggestion
 * When entities are deleted in the underlying state storage, they will stop being returned by {@link #all(Class)} and
 * {@link #get(Key)}. Instead they will be returned by {@link #allDeleted(Class)}. Agents can use this method to find deleted entities after an offline period (restarts or maintenance) where they may have missed the deletion event while offline. They can then handle those deleted entities accordingly (e.g. remove the entity from external dependencies etc). This handling
 * does not automatically remove the entities from this in memory state (or the underlying state storage). In order to
 * remove the entity, {@link #purgeDeleted(Key)} should be used. This will remove all in memory references to the
 * deleted entity and it will no longer be returned by {@link #allDeleted(Class)}.
```